libs/payments/stripe/src/lib/util/doesPriceMatchSubplatInterval.ts

```diff
@@ -33,14 +33,14 @@ const subplatIntervalToInterval = {
   },
 } satisfies Record<SubplatInterval, Interval>;
 
-export const doesPlanMatchSubplatInterval = (
-  plan: StripePlan,
+export const doesPriceMatchSubplatInterval = (
+  price: StripePrice,
   subplatInterval: SubplatInterval
 ) => {
   const stripeInterval = subplatIntervalToInterval[subplatInterval];
 
   return (
-    plan.interval === stripeInterval.interval &&
-    plan.interval_count === stripeInterval.intervalCount
+    price.recurring?.interval === stripeInterval.interval &&
+    price.recurring?.interval_count === stripeInterval.intervalCount
```

Comment:
We can update subplatIntervalToInterval to be `Record<SubplatInterval, Stripe.Price.Recurring>` and eliminate `Interval`

Reply:
For posterity - `Stripe.Price.Recurring` has properties other than `interval` and `interval_count`, and therefore the suggested change cannot be made.